tools/astarte_e2e/lib/astarte_e2e/service_notifier.ex

```diff
@@ -104,33 +147,87 @@ defmodule AstarteE2E.ServiceNotifier do
   def service_down({:call, from}, :notify_service_up, data) do
     actions = [{:reply, from, :ok}]
 
-    Email.service_up_email()
+    failure_id =
+      data
+      |> get_failure_id()
+
+    updated_data =
+      data
+      |> reset_failures_before_alert()
+      |> reset_failure_id()
+
+    Email.service_up_email(failure_id)
     |> deliver()
 
-    Logger.info("Service up. The user has been notified.", tag: "service_up_notified")
-    {:next_state, :service_up, data, actions}
-  end
+    Logger.info("Service up. The user has been notified. Last FailureID: #{failure_id}.",
+      tag: "service_up_notified"
+    )
 
-  def service_down({:call, from}, {:notify_service_down, _reason}, _data) do
-    actions = [{:reply, from, :already_notified}]
-    {:keep_state_and_data, actions}
+    {:next_state, :service_up, updated_data, actions}
   end
 
-  def service_up({:call, from}, :notify_service_up, _data) do
-    actions = [{:reply, from, :ok}]
-    {:keep_state_and_data, actions}
-  end
+  def service_down(
+        {:call, from},
+        {:notify_service_down, reason},
+        %{failures_before_alert: 0} = data
+      ) do
+    event_id = Hukai.generate("%a-%n")
```

Comment:
Let's investigate alternatives to `Hukai` since it seems it produces some not-so-production-ready combinations.
Have a look at this
https://github.com/yosriady/haiku
or, as a last resort, you could just slap something like this in a module
https://gist.github.com/coryodaniel/d5e8fa15b3d1fe566b3c3f821225936e